src/drw_base.h

```diff
@@ -108,6 +108,32 @@ enum class DebugLevel {
     Debug
 };
 
+/**
+ * Interface for debug printers.
+ *
+ * The base class is silent and ignores all debugging.
+ */
+class DebugPrinter {
+public:
+    virtual void printS(const std::string &s){(void)s;}
```

Comment:
at least one of the virtual methods should probably be defined in a .cpp file, to avoid the virtual table of the class to be output in all compilation units that include this header. See https://stackoverflow.com/questions/28786473/clang-no-out-of-line-virtual-method-definitions-pure-abstract-c-class

Reply:
Thanks, how's https://github.com/qgis/libdxfrw/pull/19/commits/0c9c7032d64aa89d5982f5c9f9c40ba3b3c8fea6?

Reply:
looks good